crates/net/network/src/metrics.rs

```diff
@@ -244,3 +246,45 @@ pub struct EthRequestHandlerMetrics {
     /// Number of received bodies requests
     pub(crate) received_bodies_requests: Counter,
 }
+
+/// Eth67 announcement metrics, track entries by TxType
+#[derive(Metrics)]
+#[metrics(scope = "network.transaction_fetcher")]
+pub struct AnnouncedTxSizeMetrics {
+    /// Histogram for tracking legacy transaction sizes
+    pub(crate) legacy: Histogram,
+
+    /// Histogram for tracking EIP-2930 transaction sizes
+    pub(crate) eip2930: Histogram,
+
+    /// Histogram for tracking EIP-1559 transaction sizes
+    pub(crate) eip1559: Histogram,
+
+    /// Histogram for tracking EIP-4844 transaction sizes
+    pub(crate) eip4844: Histogram,
+}
+
+impl AnnouncedTxSizeMetrics {
+    /// Update metrics during announcement validation, by examining each announcement entry based on
+    /// TxType
+    pub(crate) fn update_on_valid_announcement(&self, announcement: &ValidAnnouncementData) {
```

Comment:
instead of `&ValidAnnouncementData` as param, can we pass a new type `TxTypesCounter`?
```rust
struct TxTypesCounter {
    legacy: usize,
    eip2930: usize,
    eip1559: usize,
    eip4844: usize,
}
```
and then have a new function that updates `TxTypesCounter` called from here

https://github.com/paradigmxyz/reth/blob/6cf9e91144186cd301f54088b3362544903ccaa9/crates/net/network/src/transactions/validation.rs#L129-L142

?

a new instance of `TxTypesCounter` is then made for each eth68 announcement